### ISSUE TITLE
fastlane-plugin-badge 1.3 is broken

### ISSUE THREAD

**Issue by changLiuUNSW** [HazAT/badge#91]
I just upgrade from fastlane-plugin-badge 1.2 to fastlane-plugin-badge 1.3 and the badge is broken

before
<img width="89" alt="Screen Shot 2019-08-09 at 10 30 48 pm" src="https://user-images.githubusercontent.com/7379922/62779066-75f11b00-baf5-11e9-9f44-b78590c79519.png">

after
<img width="100" alt="Screen Shot 2019-08-09 at 10 29 12 pm" src="https://user-images.githubusercontent.com/7379922/62779082-80abb000-baf5-11e9-9340-3a3eeab8b8e5.png">

**Comment by changLiuUNSW** [comment]
After some research, the issue is caused by ```pango 1.44.3``` which is one of the dependencies of ```librsvg```.  It was working fine with ```pango 1.42.4```.
I cannot raise the issue on their github page https://github.com/GNOME/pango, because they disabled it.
Does anyone have the same issue and find a workaround for this issue?

**Comment by changLiuUNSW** [comment]
The workaround for now is to uninstall ```librsvg``` and use default ```graphicsmagick```

**Comment by techinpark** [comment]
@changLiuUNSW same issue. do you find any solution with using librsvg ?

**Comment by changLiuUNSW** [comment]
@techinpark no solution.

Can we raise the issue to that link?

**Comment by julianch** [comment]
> The workaround for now is to uninstall `librsvg` and use default `graphicsmagick`

Could you tell me how to use `graphicsmagick`? I uninstalled `librsvg` but I get this error: 

```
INFO [2019-08-29 18:16:36.32]: Start adding badges...
DEBUG [2019-08-29 18:16:36.32]: Trying to load image from shields.io. Timeout: 10s
DEBUG [2019-08-29 18:16:36.32]: URL: https://img.shields.io/badge/1.7-40-blue.png
ERROR [2019-08-29 18:16:36.54]: Error validating image from shields.io. Use --verbose for more info
INFO [2019-08-29 18:16:36.54]: Waiting for 10s and retry to load image from shields.io tries remaining: 5
WARN [2019-08-29 18:16:46.55]: Install RSVG to get better results for shields on top of your icon
WARN [2019-08-29 18:16:46.55]: 
WARN [2019-08-29 18:16:46.55]: Install it using (RSVG):
INFO [2019-08-29 18:16:46.55]: $ brew install librsvg
```

**Comment by PatrickDotStar** [comment]
@julianch have you install graphicsmagick? `brew install graphicsmagick`

**Comment by julianch** [comment]
> brew install graphicsmagick

@PatrickDotStar Yes, I have `graphicsmagick` installed and I'm still getting that error.

**Comment by Gremkow** [comment]
I am experiencing the same error as @julianch . I have graphicsmagick 1.3.33 installed. Sadly this does not seem to be a workaround (or might need some more configuration I am not aware of).

**Comment by techinpark** [comment]
@julianch how to you run badge?  `fastlane plugin` or `standalone` ?

**Comment by Gremkow** [comment]
`		add_badge(
			dark: true,
			shield: "Version-#{options[:app_version]}-blue",
			glob: "/path/to/png"
		)`
I use badge inside the fastfile like this.

**Comment by techinpark** [comment]
@Gremkow (cc @julianch )
In my case, i have same problem like yours. 

### Solution 
1. Open Terminal and make badge icon 
 - `badge --shield Version-#{app_version}-blue -- dark` 

2. remove add_badge line in `fastfile`

**Comment by Gremkow** [comment]
@techinpark Thanks for your advise, when trying to run your command (with a removed whitespace between -- and dark) inside the folder containing the .png, badge does not find any icons.
Anyway, we probably will need to get a solution working using the fastfile, since running manual commands is not something we want to do in our automation process, if possible.
If we find a solution to this which works for us I will update you.

**Comment by filipef101** [comment]
using librsvg solved but the badges look blury, any idea to make them less blury?

**Comment by Gremkow** [comment]
I got it working by

- manually deleting /usr/local/Cellar/pango
- manually deleting /usr/local/Cellar/librsvg

Copying /usr/local/Cellar/pango & /usr/local/Cellar/librsvg from a working machine (older formula versions: librsvg 2.44.14_1, pango 1.42.4_2) to a non working machine.

- running brew link pango
- running brew link librsvg

This might be a work around if you have access to the older version on another machine, but this is not recommend since I don't know what I'm doing with homebrew and you will probably break something and definetly wont be able to update these packages out-of-the-box later on.
Simply posting this for people that maybe REALLY need it right now and have access to old files.

**Comment by wiedem** [comment]
Easiest solution if you're using homebrew:
`brew switch pango 1.42.4_1`

**Comment by wiedem** [comment]
> This only works if you had it previously installed (and did not get removed via some kind of cleanup), right?

Yes.
In case you've already cleaned up the previous version, run the following command first:
`brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/7cf3b63be191cb2ce4cd86f4406915128ec97432/Formula/pango.rb`

**Comment by techinpark** [comment]
@wiedem Thanks for solution! 

Here is my solution. open the Terminal and type it commands.

### Solution

```sh
brew install librsvg 
```

```sh
brew unlink pango  
```

```sh
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/7cf3b63be191cb2ce4cd86f4406915128ec97432/Formula/pango.rb
```
```sh
brew switch pango 1.42.4_1 
```

- Enjoy it :tada:

**Comment by adelmojunnior** [comment]
@techinpark The 'alpha' or 'beta' label no more generate.

**Comment by aitor** [comment]
I've reported this:
* [pango based gnuplot terminals font issues on macos
](https://gitlab.gnome.org/GNOME/pango/issues/422)
* [Any scaling leads to bad text kerning on OSX
](https://gitlab.gnome.org/GNOME/librsvg/issues/520#note_627033)

**Comment by CMJunghoon** [comment]
Solution
`brew uninstall librsvg`
`brew uninstall pango`

**Comment by marcosgriselli** [comment]
Is anyone experiencing issues with `brew install librsvg`? We're using @techinpark and our CI started failing there out of the blue with no visible error.

**Comment by fruitcoder** [comment]
@marcosgriselli Happened to me a couple of days ago. I used the built in brew install step of Bitrise and some caching which I disabled and replaced everything related to librsvg with a shell script:
```
brew install librsvg
brew link librsvg

if [ -d "/usr/local/Cellar/pango/1.42.4_2" ]
then
   brew link pango
else
   brew unlink pango
   brew install --build-from-source https://raw.githubusercontent.com/Homebrew/homebrew-core/a8ac7ea5/Formula/pango.rb
fi
```

I don't know if it helps (I think it's been posted before)

**Comment by marcosgriselli** [comment]
@fruitcoder thanks! You pushed me in the right direction. I was using the script step but left the `set -e` line which was probably reporting a false error and failed the step with no reason. Removed it and the installation is working again and the badge looks perfect.

**Comment by aitor** [comment]
I can confirm that. although the reported issues are still open, pango 1.46.1 fixed the font tracking problem for us and we are using off-the-shelf dependencies configuration again for badge.

**Comment by leonardarnold** [comment]
I came across the issue today and i resolved it with:

`brew upgrade librsvg`

Thanks @aitor for the effort here

**Comment by fruitcoder** [comment]
I can confirm that the stable version of pango is working and the issue could be closed

**Comment by benkane** [comment]
I can also confirm this is no longer an issue with the latest version of librsvg on homebrew. I think this can be closed

**Comment by hahtml** [comment]
I can confirm this issue has been fixed on my side as well!

**Comment by phisakel** [comment]
brew install --build-from-source https://raw.githubusercontent.com/Homebrew/homebrew-core/a8ac7ea5/Formula/pango.rb
==> 
Error: Calling Installation of pango from a GitHub commit URL is disabled! Use 'brew extract pango' to stable tap on GitHub instead.

**Comment by benkane** [comment]
@phisakel straight up `brew install librsvg` works now and will give you a working version of pango. No need to manually install a different version

**Comment by HazAT** [comment]
This was a pango / librsvg regression that was fixed upstream in pango 1.46+. Multiple commenters in this thread (from 2020-09 onward) confirmed a plain `brew install librsvg` on a current Homebrew installs a working version. I'll close this for now — please open a new issue if you hit a fresh variant on a current install. Thanks for the detailed repro from everyone 🙏
